pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,177 @@
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+# the well-known path of the Satellite instance URL where
+# the provisioning script should be located
+PROVISIONING_SCRIPT_SUB_PATH = "/pub/katello-rhsm-consumer"
+
+
+def download_satellite_provisioning_script(satellite_url, proxy_url=None):
+    """Download provisioning script from a Red Hat Satellite instance.
+
+    Download the provisioning script from a Satellite instance and return
+    it as a string.
+
+    Satellite instances usually have self signed certificates and also some tweaks
+    are usually required in rhsm.conf to connect to a customer run Satellite instance
+    instead of to Hosted Candlepin for subscription purposes.
+
+    Each Satellite instance thus hosts a provisioning script available over plain
+    HTTP that client machines can download and execute. This script has minimal dependencies
+    and provisions the machine to be able to talk to the one given Satellite instance
+    by installing it's self signed certificates and adjusting rhsm.conf.
+
+    NOTE: As the script is downloaded over plain HTTP it is advised to ever only
+          provision machines from a Satellite instance on a trusted network, to
+          avoid the possibility of the provisioning script being tempered with
+          during transit.
+
+    :param str satellite_url: Satellite instance URL
+    :param proxy_url: proxy URL to use when fetching the script
+    :type proxy_url: str or None if not set
+    :returns: True on success, False otherwise
+    """
+    # make sure the URL starts with protocol
+    if not satellite_url.startswith("http"):
+        satellite_url = "http://" + satellite_url
+
+    # construct the URL pointing to the provisioning script
+    script_url = satellite_url + PROVISIONING_SCRIPT_SUB_PATH
+
+    log.debug("subscription: fetching Satellite provisioning script from: %s", script_url)
+
+    headers = {"user-agent": constants.USER_AGENT}
+    proxies = {}
+    provisioning_script = ""
+
+    # process proxy URL (if any)
+    if proxy_url is not None:
+        try:
+            proxy = ProxyString(proxy_url)
+            proxies = {"http": proxy.url,
+                       "https": proxy.url}
+        except ProxyStringError as e:
+            log.info("subscription: failed to parse proxy when fetching Satellite"
+                     " provisioning script %s: %s",
+                     proxy_url, e)
+
+    with util.requests_session() as session:
+        try:
+            # NOTE: we explicitly don't verify SSL certificates while
+            #       downloading the provisioning script as the Satellite
+            #       instance will most likely have it's own self signed certs that
+            #       will only be trusted once the provisioning script runs
+            result = session.get(script_url, headers=headers,
+                                 proxies=proxies, verify=False,
```

Comment:
Yes, this is certainly something we can improve in the future - there have been some discussions with the Satellite team for example, about being able to drop in a certificate via a `%pre` script, so that it is possible to fetch the provisioning script over a secure connection. Of course then you ideally also need to provide the kickstart in a way that can't be intercepted and changed by an attacker.

In any case, the default behaviour needs to stay this way for now, due to how most customer Satellite instances are configured by default. We could add an option somewhere to enforce cert checking in case there turn out to be some Satellite instances that do use certificates that the installation images trusts.

Reply:
ok, I agree that this should stay as the default and if we want to change it we have to introduce some sort of an API for that.

Reply:
Yeah, one idea was to look for the certificate in a dropdir or to provide an option to point to the certificate file. In both cases the idea is that a %pre script will drop it in place or possibly it would be part of a custom image built by Image Builder or similar.

But that's all preliminary - we will see what further discussions with the Satellite team will bring.